pydm/widgets/timeplot.py

```diff
@@ -117,15 +118,19 @@ def address(self):
     @address.setter
     def address(self, new_address: str):
         """Creates the channel for the input address for communicating with the address' plugin."""
+        if self.channel:
+            if new_address == self.channel.address:
+                return
+            self.channel.disconnect()
+
         if not new_address:
             self.channel = None
             return
-        elif self.channel and new_address == self.channel.address:
-            return
 
         self.channel = PyDMChannel(
             address=new_address, connection_slot=self.connectionStateChanged, value_slot=self.receiveNewValue
         )
+        self.channel.connect()
```

Comment:
This line results in this error for me, and sometimes causes the plot to not connect to the PV at all:

`
QObject::connect: Cannot connect Connection::connection_state_signal(bool) to (nullptr)::connectionStateChanged(bool)`

Can reproduce by running the example in `examples/timeplot/timeplot.ui`

Connections for plots look to be made in baseplot here:

https://github.com/slaclab/pydm/blob/c47529c79306b73b157e79bfb07cf66bab74110d/pydm/widgets/baseplot.py#L839-L841

Not sure if there's functionality gained by moving it earlier to the setter (beside making it clearer where the connection happens), but would need to fix the error and probably update the other plots too to avoid the double connect() call.

Reply:
I hadn't gotten that error, but I'll try to replicate it and fix it.

I think it didn't come up for me because of how I'm using ArchiveCurveItems in the Archive Viewer: we create a blank ArchiveCurveItem with no channel and the user enters one after initialization.

This use case is also why I included connecting the channel in the address setter. Since we are changing the channel after the fact, we want to disconnect the old one and connect the new one in the address setter rather than on initialization. Users may also set the address multiple times (e.g. entering a typo the first time) and creating a new curve object every time would be a hassle.

Reply:
This error was being caused by trying to make these connections before calling `super().__init__()` so I moved that up in `PyDMTimePlot.__init__()`. This change worked fine in my testing, hope it does for you too.